xclim/indices/_threshold.py

```diff
@@ -1928,6 +1929,59 @@ def heat_wave_index(
     return to_agg_units(out, tasmax, "count")
 
 
+@declare_units(tasmax="[temperature]", thresh="[temperature]")
+def heat_wave_magnitude(
+    tasmax: xarray.DataArray,
+    thresh: Quantified = "25.0 degC",
+    window: int = 3,
+    freq: str = "YS",
+    op: str = ">",
+    resample_before_rl: bool = True,
+) -> xarray.DataArray:
+    """Heat wave magnitude index.
+
+    Magnitude of the most intensive heat wave event as sum of differences between tasmax
+    and the given threshold for Heat Wave days, defined as three or more consecutive days
+    over the threshold.
+
+    Parameters
+    ----------
+    tasmax : xarray.DataArray
+        Maximum daily temperature.
+    thresh : xarray.DataArray
+        Threshold temperature on which to designate a heatwave.
+    window : int
+        Minimum number of days with temperature above threshold to qualify as a heatwave.
+    freq : str
+        Resampling frequency.
+    op : {">", ">=", "gt", "ge"}
+        Comparison operation. Default: ">".
+    resample_before_rl : bool
+        Determines if the resampling should take place before or after the run
+        length encoding (or a similar algorithm) is applied to runs.
+
+    Returns
+    -------
+    DataArray, [time]
+        Heat wave magnitude.
+    """
+    thresh = convert_units_to(thresh, tasmax)
+    # condition where over threshold is fulfilled
+    cond = compare(tasmax, op, thresh, constrain=(">", ">="))
+    # difference for heatwave magnitude calculation
+    over_values = (tasmax - thresh).where(cond, 0)
+
+    out = rl.resample_and_rl(
+        over_values,
+        resample_before_rl,
+        rl.windowed_max_run_sum,
+        window=window,
+        freq=freq,
+    )
+    out.attrs["units"] = tasmax.attrs["units"]
+    return out
```

Comment:
```suggestion
    return to_agg_units(out, tasmax, op="integral")
```
Shouldn't the output be "K d" ? A sum of temperatures doesn't mean much, but an integral does, no ? 
And xclim (in the last few versions) has made the decision to avoid using "degC day" as it is ambiguous.

Reply:
Changing the units in https://github.com/Ouranosinc/xclim/pull/1926/files#diff-5d97458806cde0bbdc29e9a9b0e3209999b8f6d9b12b90ab9ac499a956d8ddd2R222 from `degC` to `K d` combined with this suggestion has indeed fixed the failing test. Thank you!